compiler/src/dotty/tools/dotc/transform/MacroAnnotations.scala

```diff
@@ -130,6 +134,20 @@ class MacroAnnotations:
     else if annotated.isClass && annotated.owner.is(Package) /*&& !sym.isClass*/ then
       report.error(i"macro annotation can not add top-level ${sym.showKind}. $annot tried to add $sym.", annot.tree)
 
+  private def enterMissingSymbols(tree: DefTree)(using Context) = new TreeTraverser {
+    def traverse(tree: tpd.Tree)(using Context): Unit = tree match
+      case tdef @ TypeDef(_, template: Template) =>
+        //for tree <- template.constr :: template.body do
+        val isSymbolInDecls = tdef.symbol.asClass.info.decls.toList.toSet
+        for tree <- template.body do
+          if tree.symbol.owner != tdef.symbol then
+            report.error(em"Macro added a definition with the wrong owner - ${tree.symbol.owner} - ${tdef.symbol} in ${tree.source}")
```

Comment:
We should have a test that produces this error. This test should have a checkfile.

Reply:
I don't think we need this check anymore. When running the `wrong-owner` test, I have the following result in `testlog`

> -- Error: tests/neg-macros/wrong-owner/Test_2.scala:7:6 ----------------------------------------------------------------
> 5 |@experimental
> 6 |@wrongOwner
> 7 |class Foo
>   |^
>   |Malformed tree was found while expanding macro with -Xcheck-macros.
>   |               |The tree does not conform to the compiler's tree invariants.
>   |               |
>   |               |Macro was:
>   |               |@scala.annotation.internal.SourceFile("tests/neg-macros/wrong-owner/Test_2.scala") @wrongOwner @scala.annotation.experimental class Foo()
>   |               |
>   |               |The macro returned:
>   |               |@scala.annotation.internal.SourceFile("tests/neg-macros/wrong-owner/Test_2.scala") @wrongOwner @scala.annotation.experimental class Foo() {
>   |  override def toString(): java.lang.String = "Hello from macro"
>   |}
>   |               |
>   |               |Error:
>   |               |assertion failed: bad owner; method toString has owner class String, expected was class Foo
>   |owner chain = method toString, class String, package java.lang, package java, package <root>, ctxOwners = class Foo, class Foo, package <empty>, package <empty>, package <empty>, package <root>, package <root>, package <root>, package <root>, package <root>, package <root>, <none>, <none>, <none>, <none>, <none>
>   |               |
>   |stacktrace available when compiling with `-Ydebug`
>   |               |
> Macro added a definition with the wrong owner - class String - class Foo in tests/neg-macros/wrong-owner/Test_2.scala
> 2 errors found

Reply:
Fixed. The issue was that tests under the `/tests/neg-macros` run with the `-Xcheck-macros` which itself trigger the second error. We agreed with @nicolasstucki to keep both error messages in the sense that if the flag is enabled, the more developed error message will be displayed.

Reply:
Actually, maybe we should add this information in the new error message. e.g "More information available under with the `-Xcheck-macros`"